pkg/option/config.go

```diff
@@ -3087,8 +3091,14 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 	if c.TunnelPort == 0 {
 		switch c.Tunnel {
 		case TunnelDisabled:
-			// tunnel might still be used by eg. EgressGW
-			c.TunnelPort = defaults.TunnelPortVXLAN
+			// tunnel might still be used by eg. DSR with Geneve dispatch or EgressGW
+			if (c.EnableNodePort || c.KubeProxyReplacement == KubeProxyReplacementStrict) &&
+				c.NodePortMode == NodePortModeDSR &&
+				c.LoadBalancerDSRDispatch == DSRDispatchGeneve {
+				c.TunnelPort = defaults.TunnelPortGeneve
```

Comment:
Does this mean that egress gateway is incompatible with this new mode? Do we have checks to validate this during startup?

Reply:
(I didn't see such checks, but if there are checks for egress gateway vs. DSR geneve dispatch, feel free to point them out)

Reply:
EgressGW is compatible with DSR Geneve dispatch. The reason for this change is that we need to set up `cilium_geneve` with port 6081 in conjunction with the direct routing, EgressGW, and DSR Geneve dispatch. Otherwise Cilium set up port 8472 (vxlan port)

I have added an e2e test which checks the compatibility.
https://github.com/cilium/cilium/pull/23890/files#diff-afcd34222934f6f76dea34fd140bc589c4ef8325ba66c82f3458e1112b920622R487-R503

Reply:
Yes, the egress gateway traffic goes over `cilium_geneve` port 6081 as well. 

> If we are intending to support both features at once over two different protocols, then I don't see how we can have a single c.TunnelPort configuration option to support both.

Do you mean Geneve and vxlan by different protocols? We only use Geneve for both EgressGW and DSR, and it's working.

Reply:
Oh, I didn't realize that, I assumed that egress gateway just uses VXLAN. The previous code unconditionally set the port to the VXLAN port. Have we been using the wrong port the whole time?

Reply:
Previously the egress gateway always uses VXLAN in the direct routing, but with the DSR Geneve, we only use `cilium_geneve` because we can't use both `cilium_vxlan` and `cilium_geneve` in the current implementation. That's why this change is needed.
https://github.com/cilium/cilium/pull/23890/files#diff-3bd10cddebdc5669fc7a472e02c78702a9bc59fee30d502d1cc87a84a58439d8R349-R354